diy_view/src/main/java/scut/carsonho/diyview/SuperEditText.java

```diff
@@ -0,0 +1,422 @@
+
+    // Delete icon
+    private Element icdelete;
+
+    // The starting point of the Delete icon (x,y), the width and height of the Delete icon (px)
+    int deletex;
+    int deletey;
+    int deletewidth;
+    int deleteheight;
+
+    // Icon resource ID on the left (click & no click)
+    int icleftclickResId;
+    int icleftunclickResId;
+
+    // Icon on the left (clicked & not clicked)
+    private Element icleftclick;
+    private Element icleftunclick;
+
+    //  The starting point of the icon on the left (x,y), the width and height of the icon on the left (px)
+    int leftx;
+    int lefty;
+    int leftwidth;
+    int leftheight;
+
+    // Cursor
+    int cursor;
+
+    // Dividing line variable
+    // When clicked & color not clicked
+    private int lineColorclick;
+    private int lineColorunclick;
+    private int color;
+    private int linePosition;
+
+    /**
+     * SuperEditText Constructor.
+     *
+     *  @param context - context for SuperEditText constructor
+     *
+     */
+    public SuperEditText(Context context) {
+        super(context);
+        addTextObserver(this);
+        setFocusChangedListener(this);
+        setTouchEventListener(this);
+        addDrawTask(this);
+    }
+
+    /**
+     * SuperEditText Constructor.
+     *
+     *  @param context - context for SuperEditText constructor
+     *  @param attrs - attributes
+     *
+     */
+    public SuperEditText(Context context, AttrSet attrs) {
+        super(context, attrs);
+        init(context, attrs);
+        addTextObserver(this);
+        setFocusChangedListener(this);
+        setTouchEventListener(this);
+        addDrawTask(this);
+    }
+
+    /**
+     * SuperEditText Constructor.
+     *
+     *  @param context - context for SuperEditText constructor
+     *  @param attrs - attributes
+     *  @param defStyleAttr - defStyle attribute
+     *
+     */
+    public SuperEditText(Context context, AttrSet attrs, String defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init(context, attrs);
+        addTextObserver(this);
+        setFocusChangedListener(this);
+        setTouchEventListener(this);
+        addDrawTask(this);
+    }
+
+    /**
+     *  Step 1: Initialize the properties.
+     */
+    private void init(Context context, AttrSet attrs) {
+        try {
+            // Initialize the icon on the left (clicked & not clicked)
+            // a. Click the icon on the left side of the status
+            // 1. Get resource ID
+            icleftclickResId = attrs.getAttr("ic_left_click").isPresent()
+                    ? attrs.getAttr("ic_left_click").get().getIntegerValue()
+                    : ResourceTable.Media_ic_left_click;
+            // 2. Get icon id based on getElementFromResId
+            icleftclick = getElementFromResId(getContext(), icleftclickResId).get();
+            // 3. Set icon size
+            leftx = AttrUtils.getIntFromAttr(attrs, "left_x", 0);
+            // 3. Set icon size
```

Comment:
//3. Set icon size
This comment need not be repeated several times. Keep only one instance of this comment, it will be implied for all the following statements too

Reply:
Done